include/seastar/http/client.hh

```diff
@@ -259,33 +258,13 @@ public:
      * \param req -- request to be sent
      * \param handle -- the response handler
      * \param expected -- the optional expected reply status code, default is std::nullopt
+     * \param as -- abort source that aborts the request
      *
      * Note that the handle callback should be prepared to be called more than once, because
      * client may restart the whole request processing in case server closes the connection
      * in the middle of operation
      */
-    future<> make_request(request&& req, reply_handler&& handle, std::optional<reply::status_type>&& expected = std::nullopt);
-
-    /**
-     * \brief Send the request and handle the response (abortable)
-     *
-     * Same as previous method, but aborts the request upon as.request_abort() call
-     *
-     * \param req -- request to be sent
-     * \param handle -- the response handler
-     * \param as -- abort source that aborts the request
-     * \param expected -- the optional expected reply status code, default is std::nullopt
-     */
-    future<> make_request(request&& req, reply_handler&& handle, abort_source& as, std::optional<reply::status_type>&& expected = std::nullopt);
-
-    /**
-     * \brief Send the request and handle the response, same as \ref make_request()
-     *
-     *  @attention Note that the method does not take the ownership of the
-     * `request and the `handle`, it caller's responsibility the make sure they
-     * are referencing valid instances
-     */
-    future<> make_request(request& req, reply_handler& handle, std::optional<reply::status_type> expected = std::nullopt);
+    future<> make_request(request&& req, reply_handler&& handle, std::optional<reply::status_type>&& expected = std::nullopt, abort_source* as = nullptr);
```

Comment:
Shouldn't you keep the old API, perhaps deprecated?

The change from value parameter to rvalue reference can break callers.

Reply:
Maybe you can pass a request_control struct that contains expected and abort_source, so we can grow more control parameters later.

Reply:
> Shouldn't you keep the old API, perhaps deprecated?

It's still `namespace experimental`, so I'd take advantage of it

Reply:
> Maybe you can pass a request_control struct that contains expected and abort_source, so we can grow more control parameters later.

But we can as well grow the arguments list. So far there are two "controls" which is probably not enough for the dedicated controller struct

Reply:
@avikivity , reminder ping

Reply:
Sorry. Wouldn't we also expect an optional timeout in the future?

Reply:
With the `abort_source` at hand it's [arguable](https://github.com/scylladb/seastar/pull/1889#issuecomment-1777915851) that dedicated timeout parameter will be needed